aiida/cmdline/commands/cmd_export.py

```diff
@@ -141,31 +142,45 @@ def create(
 
 @verdi_export.command('migrate')
 @arguments.INPUT_FILE()
-@arguments.OUTPUT_FILE()
+@arguments.OUTPUT_FILE(required=False)
 @options.ARCHIVE_FORMAT()
 @options.FORCE(help='overwrite output file if it already exists')
+@click.option('-i', '--in-place', is_flag=True, help='Migrate the archive in place, overwriting the original file.')
 @options.SILENT()
 @click.option(
     '-v',
     '--version',
     type=click.STRING,
     required=False,
     metavar='VERSION',
-    help='Specify an exact archive version to migrate to. By default the most recent version is taken.'
+    # Note: Adding aiida.tools.EXPORT_VERSION as a default value explicitly would result in a slow import of
+    # aiida.tools and, as a consequence, aiida.orm. As long as this is the case, better determine the latest export
+    # version inside the function when needed.
+    help='Archive format version to migrate to (defaults to latest version).',
 )
-def migrate(input_file, output_file, force, silent, archive_format, version):
+def migrate(input_file, output_file, force, silent, in_place, archive_format, version):
     # pylint: disable=too-many-locals,too-many-statements,too-many-branches
     """Migrate an export archive to a more recent format version."""
     import tarfile
     import zipfile
 
     from aiida.common import json
     from aiida.common.folders import SandboxFolder
-    from aiida.tools.importexport import EXPORT_VERSION, migration, extract_zip, extract_tar, ArchiveMigrationError
+    from aiida.tools.importexport import migration, extract_zip, extract_tar, ArchiveMigrationError, EXPORT_VERSION
 
     if version is None:
         version = EXPORT_VERSION
 
+    if in_place:
+        if output_file:
+            echo.echo_critical('output file specified together with --in-place flag')
+        tempdir = tempfile.TemporaryDirectory()
```

Comment:
not having this in a context manager, will not trigger the automatic cleanup. It would be good to try and use `with tempfile.TemporaryDirectory as tempdir` and not depend on your own `tempdir.cleanup` call. I realize that this would mean moving all code below within the context, but maybe you can refactor/reshuffle the code a bit to have the opening of the context as short as possible.

Reply:
I think the trouble with using a context manager is that it's only needed in the `in_place` branch, and not the other branch.

In my (admittedly limited) testing, the temporary directory will be destroyed when the Python interpreter exits at the latest.

Reply:
> In my (admittedly limited) testing, the temporary directory will be destroyed when the Python interpreter exits at the latest.

I still think that it might be nicer to use a context manager, despite it only being used in a particular branch of code. Part of the reason why it would be cumbersome in the current implementation is because the code is a bit messy. The whole code about migration (unpacking, checking, migration, repacking) should be done in some utility. Once that is done, having the context manager would not be so problematic.

But as long as the folder is cleaned after interpreter, I think it is ok for now

Reply:
> But as long as the folder is cleaned after interpreter, I think it is ok for now

That's observed behavior, not sure if it's _guaranteed_ behavior, since "It is not guaranteed that ``__del__()`` methods are called for objects that still exist when the interpreter exits." (https://docs.python.org/3/reference/datamodel.html#object.__del__).

Reply:
Another option would be using a `contextlib.nullcontext` for the `in_place=False` branch - creating the context manager object conditionally, and then entering it later.

Of course, if the code can be cleaned up to not need that, even better.

Reply:
After some prodding: the cleanup is implemented with [`weakref.finalize`](https://docs.python.org/3/library/weakref.html#weakref.finalize), not a `__del__` method. I'm still not 100% sure about it, but I think unless things go absolutely haywire it should always be cleaned.

Reply:
Hmm, still not ideal, but since this would require more refactoring to be effective, I will leave this for now.